### PR TITLE
[mono] Improve how EnumBuilder handles type attributes

### PR DESCRIPTION
**Pull request by alexischr**

### CODE REVIEW

**Comment by danmosemsft** [comment]
Please help add (exactly one) `area-XXX` label when the bot does not add one - that will help train it.

**Comment by alexischr** [comment]
This will still not still not pass the DefineEnum test, since Mono adds the `RTSpecialName` attribute to fields prematurely (before `CreateTypeInfo` is used), causing a lot of tests to fail including this one on a later assert. I am looking on how to fix that.

File: src/mono/netcore/CoreFX.issues.rsp [hunk 1]
```diff
@@ -234,7 +234,6 @@
 -nomethod System.Reflection.Emit.Tests.MethodBuilderToString.ToString_NameAttributeAndSignatureSetSet
 -nomethod System.Reflection.Emit.Tests.MethodBuilderToString.ToString_NonGenericMethod
 -nomethod System.Reflection.Emit.Tests.MethodBuilderToString.ToString_GenericMethod
--nomethod System.Reflection.Emit.Tests.ModuleBuilderDefineEnum.DefineEnum
```
Comment (akoeplinger) [inline src/mono/netcore/CoreFX.issues.rsp]: note that this file was removed so you need to rebase and remove the `ActiveIssue` here instead:

https://github.com/dotnet/runtime/blob/6a33d96168c2f7ae3f9e803fa646d9c3bb821488/src/libraries/System.Reflection.Emit/tests/ModuleBuilder/ModuleBuilderDefineEnum.cs#L33-L35

**Comment by alexischr** [comment]
Closing, accidentally pushed branch on main repo instead of my fork

**Pull request closed without merging**